### PR TITLE
Add maestro dependency flow implementation

### PR DESCRIPTION
**Pull request by alexperovich**

### CODE REVIEW

File: src/Maestro/Maestro.Data/Migrations/20180811022249_initial.cs [hunk 1]
```diff
@@ -409,6 +425,11 @@ protected override void Up(MigrationBuilder migrationBuilder)
                 name: "IX_Subscriptions_ChannelId",
                 table: "Subscriptions",
                 column: "ChannelId");
+
+            migrationBuilder.CreateIndex(
+                name: "IX_Subscriptions_LastAppliedBuildId",
```
Comment (jcagme) [inline src/Maestro/Maestro.Data/Migrations/20180811022249_initial.cs]: What does `IX` stands for?

File: src/Maestro/Maestro.Data/Migrations/BuildAssetRegistryContextModelSnapshot.cs [hunk 2]
```diff
@@ -1,15 +1,12 @@
-// Licensed to the .NET Foundation under one or more agreements.
```
Comment (jcagme) [inline src/Maestro/Maestro.Data/Migrations/BuildAssetRegistryContextModelSnapshot.cs]: No header in auto-generated?

File: src/Maestro/Maestro.Data/Models/Subscription.cs [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+namespace Maestro.Data.Models
+{
+    public class Subscription
+    {
+        [Key]
+        [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
+        public Guid Id { get; set; }
+
+        public int ChannelId { get; set; }
+        public Channel Channel { get; set; }
```
Comment (jcagme) [inline src/Maestro/Maestro.Data/Models/Subscription.cs]: Nit: add line before so the patters in conserved

File: src/Maestro/ServiceFabricMocks/MockReliableDictionary.cs [hunk 4]
```diff
@@ -0,0 +1,340 @@
...
+        public long Count => dictionary.Count();
+
+        public Task AddAsync(ITransaction tx, TKey key, TValue value)
+        {
+            if (!dictionary.TryAdd(key, value))
+            {
+                throw new InvalidOperationException("key already exists: " + key);
+            }
+
+
```
Comment (jcagme) [inline src/Maestro/ServiceFabricMocks/MockReliableDictionary.cs]: Remove extra line

File: src/Maestro/Maestro.Data/Migrations/20180811022249_initial.cs (hunk 1 above)
Comment (alexperovich) [reply src/Maestro/Maestro.Data/Migrations/20180811022249_initial.cs]: Index. This is part of the autogenerated migrations from Entity Framework.

File: src/Maestro/Maestro.Data/Migrations/BuildAssetRegistryContextModelSnapshot.cs (hunk 2 above)
Comment (alexperovich) [reply src/Maestro/Maestro.Data/Migrations/BuildAssetRegistryContextModelSnapshot.cs]: They got deleted when I regenerated things. I will put them back.

File: src/Microsoft.DotNet.Darc/src/DarcLib/GitHubClient.cs [hunk 5]
```diff
@@ -61,12 +61,23 @@ public async Task CreateDarcBranchAsync(string repoUri, string branch)
...
-            GitHubRef githubRef = new GitHubRef($"refs/heads/darc-{branch}", latestSha);
+            string gitRef = $"refs/heads/darc-{branch}";
+            GitHubRef githubRef = new GitHubRef(gitRef, latestSha);
             HttpResponseMessage response = null;
 
             try
             {
                 response = await this.ExecuteGitCommand(HttpMethod.Get, $"repos/{ownerAndRepo}branches/darc-{branch}", _logger);
+
+                await this.ExecuteGitCommand(
```
Comment (jcagme) [inline src/Microsoft.DotNet.Darc/src/DarcLib/GitHubClient.cs]: You could use githubRef in line 65 and just do Force = true. Then just set the body ` body = JsonConvert.SerializeObject(githubRef, _serializerSettings);` and call `ExecuteGitCommand` with it

File: src/Microsoft.DotNet.ServiceFabric.ServiceHost/FixedApplicationInsigthtsLoggerProvider/FixedApplicationInsightsLogger.cs [hunk 6]
```diff
@@ -0,0 +1,97 @@
...
+using System.Linq;
+using System.Text.RegularExpressions;
+using Microsoft.ApplicationInsights;
+using Microsoft.ApplicationInsights.DataContracts;
+using Microsoft.Extensions.Logging;
+using Newtonsoft.Json;
+
+namespace Microsoft.DotNet.ServiceFabric.ServiceHost
+{
+    public class FixedApplicationInsightsLogger : ILogger
```
Comment (jcagme) [inline src/Microsoft.DotNet.ServiceFabric.ServiceHost/FixedApplicationInsigthtsLoggerProvider/FixedApplicationInsightsLogger.cs]: Curious, what does this fix?

Comment (alexperovich) [reply src/Microsoft.DotNet.ServiceFabric.ServiceHost/FixedApplicationInsigthtsLoggerProvider/FixedApplicationInsightsLogger.cs]: It fixes the ILogger.BeginScope implementation. The application insights provider doesn't implement it at all.

**Pull request merged**